src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -146,6 +146,22 @@ public class ConsensusManager : IConsensusManager
             nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark, 1000);
         }
 
+        /// <inheritdoc />
+        public ConsensusManagerState State()
+        {
```

Comment:
This needs to have the correct locks I would imagine, otherwise tests might throw.